pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -114,6 +114,12 @@ func (pdev *podDevices) podDevices(podUID, resource string) sets.Set[string] {
 func (pdev *podDevices) containerDevices(podUID, contName, resource string) sets.Set[string] {
 	pdev.RLock()
 	defer pdev.RUnlock()
+	return pdev.listContainerDevices(podUID, contName, resource)
+}
+
+// Returns list of device Ids allocated to the given container for the given resource without RLock.
+// Returns nil if we don't have cached state for the given <podUID, contName, resource>.
+func (pdev *podDevices) listContainerDevices(podUID, contName, resource string) sets.Set[string] {
```

Comment:
let's call it `listContainerDevicesUnlocked` . Please add a comment line stating this function is meant to be used as helper and should not be called directly

Reply:
Updated.